aspnetcore/fundamentals/http-context/samples/6.x/HttpContextInBackgroundThread/EmailService.cs

```diff
@@ -1,31 +1,41 @@
 namespace HttpContextInBackgroundThread;
 
+public interface IEmailService
+{
+    void SendEmail(string email);
+}
+
 public class EmailService : IEmailService
 {
     private readonly IHttpContextAccessor _httpContextAccessor;
     private readonly ILogger<EmailService> _logger;
-    private const string? BackgroundCorrelationId = "background-correlation-id";
+    private const string UserAgent = "Unknown";
 
-    public EmailService(IHttpContextAccessor httpContextAccessor, ILogger<EmailService> logger)
+    public EmailService(IHttpContextAccessor httpContextAccessor,
+                                     ILogger<EmailService> logger)
     {
         _httpContextAccessor = httpContextAccessor;
         _logger = logger;
     }
 
     public void SendEmail(string email)
     {
-        // Services should account for the possibility of HttpContext being null if not called from a request thread.
-        // e.g. When a service is called from a BackgroundWorker the HttpContext is null.
-        // The service should either throw or gracefully handle this.
+        // Services should account for the possibility of HttpContext being null
+        // if not called from a request thread. For example, when a service is 
+        // called from a BackgroundWorker, the HttpContext is null.
+        // The service should either throw or gracefully handle a null HttpContext.
 
-        var correlationId = _httpContextAccessor.HttpContext?.Request.Headers["X-Correlation-Id"].ToString() ?? BackgroundCorrelationId;
+        var request = _httpContextAccessor.HttpContext?.Request;
+        var userAgent = request?.Headers["user-agent"].ToString()
+                                         ?? UserAgent;
 
-        _ = SendEmailCoreAsync(correlationId);
+        _ = SendEmailCoreAsync(userAgent);
     }
 
-    private async Task SendEmailCoreAsync(string? correlationId)
+    private async Task SendEmailCoreAsync(string userAgent)
     {
-        _logger.LogInformation($"Email sent with correlation id: {correlationId}");
+        _logger.LogInformation($"Email sent detected user agent: {userAgent}");
+
         await Task.CompletedTask;
```

Comment:
This will introduce a warning that nothing's being `await`ed, but I think that's slightly better for this sample code as `await Task.CompletedTask` isn't typically useful:

```suggestion
        // ...
```